src/clj/puppetlabs/services/jruby_pool_manager/impl/jruby_internal.clj

```diff
@@ -41,16 +33,24 @@
     :force RubyInstanceConfig$CompileMode/FORCE
     :off RubyInstanceConfig$CompileMode/OFF))
 
+(schema/defn ^:always-validate get-compat-version
+  "Get the JRuby compatibility version to use for all ruby components, e.g. the
+  master service and CLI tools."
+  [compat-version :- jruby-schemas/SupportedJRubyCompatVersions]
+  (case compat-version
+    "1.9" (CompatVersion/RUBY1_9)
+    "2.0" (CompatVersion/RUBY2_0)))
```

Comment:
In this case I don't think it will give a more user-friendly error message, since it should never get here - there will either be a schema error from the `initialize-config` schema, or from the schema error. However, I agree that having a default case makes sense, in case the schema changes or anything like that.